spec/requests/api/v2/people_controller_index_spec.rb

```diff
@@ -0,0 +1,175 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Api::V2::PeopleController do
+  let(:supplier) { create(:supplier) }
+  let!(:application) { create(:application, owner_id: supplier.id) }
+  let!(:access_token) { create(:access_token, application: application).token }
+  let(:headers) { { 'CONTENT_TYPE': content_type }.merge('Authorization' => "Bearer #{access_token}") }
+  let(:response_json) { JSON.parse(response.body) }
+  let(:content_type) { ApiController::CONTENT_TYPE }
+
+  describe 'GET /people' do
+    let(:schema) { load_yaml_schema('get_people_responses.yaml', version: 'v2') }
+    let!(:people) { create_list :person, 2 }
+    let(:params) { {} }
+
+    context 'when successful' do
+      context 'when there are no params' do
+        before { get '/api/v2/people', params: params, headers: headers }
+
+        it_behaves_like 'an endpoint that responds with success 200'
+
+        # TODO: think about this!
+        it 'returns correct attributes' do
+          expect(response_json['data'].first['attributes']).to include('last_name', 'first_names', 'date_of_birth', 'gender_additional_information')
+        end
+
+        it 'returns the correct number of people' do
+          expect(response_json['data'].count).to eq(2)
+        end
+      end
+
+      describe 'filtering results by police_national_computer' do
+        let!(:person) { create(:person, police_national_computer: 'AB/1234567') }
+        let(:filters) do
+          {
+            bar: 'bar',
+            police_national_computer: 'AB/1234567',
+            foo: 'foo',
+          }
+        end
+        let(:params) { { filter: filters } }
+
+        before { get '/api/v2/people', params: params, headers: headers }
+
+        it 'returns the correct number of people' do
+          expect(response_json['data'].size).to eq(1)
+        end
+
+        it 'returns the person that matches the filter' do
+          expect(response_json).to include_json(data: [{ id: person.id }])
+        end
+      end
+
+      describe 'filtering results by multiple filters' do
+        let!(:person) do
+          create(:person, criminal_records_office: 'CRO0105d', prison_number: 'D00001dZZ', police_national_computer: 'AB/00001d')
+        end
+        let(:filters) do
+          {
+            bar: 'bar',
+            criminal_records_office: 'CRO0105d',
+            police_national_computer: 'AB/00001d',
+            prison_number: 'D00001dZZ',
+            foo: 'foo',
+          }
+        end
+        let(:params) { { filter: filters } }
+
+        before { get '/api/v2/people', params: params, headers: headers }
+
+        it 'returns the correct number of people' do
+          expect(response_json['data'].size).to eq(1)
+        end
+
+        it 'returns the person that matches the filter' do
+          expect(response_json).to include_json(data: [{ id: person.id }])
+        end
+      end
+
+      xdescribe 'filtering results by multiple values per filter'
```

Comment:
guess this is still wip ?

Reply:
yea still working through this, please hold off the review